src/stan_math_backend/Expression_gen.ml

```diff
@@ -175,6 +175,8 @@ and gen_misc_special_math_app f =
           else pf ppf "%s(@[<hov>%a@])" f (list ~sep:comma pp_expr) es )
   | f when f = string_of_internal_fn FnLength ->
       Some (fun ppf -> gen_fun_app ppf "stan::length")
+  | f when f = string_of_internal_fn FnNegInf ->
```

Comment:
Am I right in thinking that you are still missing
```
  | FnMakeArray
  | FnMakeRowVec
  | FnReadData
  | FnReadParam
  | FnWriteParam
  | FnConstrain
  | FnUnconstrain
  | FnCheck
  | FnPrint
  | FnReject
```
here?

Reply:
Sort of - you have found a code smell. There is a weird conceptual mismatch here where some of these functions will be taken care of in other places. But yeah there are also some missing. 

The key issue that maybe we should talk about at a meeting is that I was thinking that since `negative_infinity` is a StanLib function we should label FnNegInf as `StanLib` when we create it from the compiler. But perhaps the thing to actually do is just use the actual string `negative_infinity` and not create a `FnVariant` for it? OR to call it `CompilerInternal` and pretend that it doesn't have a StanLib equivalent. I'm not sure. What do you all think? I'm leaning towards one of those two options now - having these special variants sometimes labeled StanLib and sometimes CompilerInternal is not helpful.

Reply:
Ah, I see what you mean. So I guess all the CompilerInternal ones are handled elsewhere and this is just supposed to catch the cases of FnNegInf and FnLength where they are StanLib?

Reply:
Yeah :< Seems bad now that we're actually using it.

Reply:
I'll make a github issue to deal with this at some point in the future.